src/components/structures/RoomView.js

```diff
@@ -54,11 +54,15 @@ module.exports = React.createClass({
     propTypes: {
         ConferenceHandler: React.PropTypes.any,
 
-        roomId: React.PropTypes.string.isRequired,
-
-        // if we are referring to this room by a given alias (e.g. in the URL), track it.
-        // useful for joining rooms by alias correctly (and fixing https://github.com/vector-im/vector-web/issues/819)
-        roomAlias: React.PropTypes.string,
+        // the ID for this room (or, if we don't know it, an alias for it)
+        //
+        // XXX: if this is an alias, we will display a 'join' dialogue,
+        // regardless of whether we are already a member, or if the room is
+        // peekable. Currently there is a big mess, where at least four
+        // different components (RoomView, MatrixChat, RoomDirectory,
+        // SlashCommands) have logic for turning aliases into rooms, and each
+        // of them do it differently and have different edge cases.
```

Comment:
As discussed via PM: this is really just making the existing situation more explicit.

We _do_ instantiate `RoomView` with an alias (for instance when you type an alias into the directory); but previously in that case the alias would have been stuffed into `props.roomId`, causing all manner of breakage. In an ideal world, we'd make this better, (and yes, I'm leaning towards RoomView being in charge of resolving aliases to rooms) but I'm hoping to avoid fixing all of the alias->room flows right now.